### DISCUSSION TITLE
How to tell firmware version

### DISCUSSION THREAD

**Discussion by Axelaar** [Arnold-n/P1P2MQTT#90] [General]
I have not been able to figure out yet how to tell which firmware version I have (ESP and/or ATmega), and hence do not know when it is time to flash new firmware on the P1P2 bridge.  I have found hwID topics for the ESP and ATmega interfaces - maybe I am overlooking which topic contains the firmware version?

**Comment by Arnold-n** [comment]
In future the version will be available as separate topics/entities (major/minor/patch for both CPUs, and brand/model for the supported system, though I don't think I will make these discoverable in HA). For now you can enter the 'V' command via telnet or MQTT - the version numbers will be visible in your telnet session or in the P1P2/S/# topic.

**Reply by Axelaar** [reply]
Thanks for the quick response!  That is an easy way to find out which versions are installed.
Interestingly enough the response included a 'brown-out' message.  It may have to do with the fact that the P1P2 bridge uses the ethernet connection, and is entirely powered by the P1P2 bus.
Just for info: That section of the response was:
P1P2/S/078 * [ESP] To ATmega: ->V<-
P1P2/S/078 * [MON] 2023-11-21_21:37:54 Verbose 3
P1P2/S/078 * [MON] 2023-11-21_21:37:54 P1P2Monitor-v0.9.42
P1P2/S/078 * [MON] 2023-11-21_21:37:54 Compiled Sep 27 2023 12:49:03 +control NEWP1P2LIB E-series
P1P2/S/078 * [MON] 2023-11-21_21:37:54 Reset cause: MCUSR=5 (brown-out-detected) (power-on-reset)
P1P2/S/078 * [MON] 2023-11-21_21:37:54 P1P2-ESP-interface ATmegaHwID v1.2

**Comment by Arnold-n** [comment]
When the interface is powered-up (after a Daikin restart or power-up), it usually reports a power-on-reset and a brown-out-detection together. This does not mean a later brown-out detection during real operation. The message can be reset by a software ATmega reset ('K') or via the reset input ('A').
